fendermint/vm/message/src/chain_message.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+use serde::{Deserialize, Serialize};
+
+use crate::signed_message::SignedMessage;
+
+/// The different kinds of messages that can appear in blocks, ie. the transactions
+/// we can receive from Tendermint through the ABCI.
+///
+/// Unlike Filecoin, we don't have `Unsigned` messages here. In Filecoin, the messages
+/// signed by BLS signatures are aggregated to the block level, and their original
+/// signatures are stripped from the messages, to save space. Tendermint Core will
+/// not do this for us (perhaps with ABCI++ Vote Extensions we could do it), though.
+#[derive(Clone, Debug, Serialize, Deserialize, Hash)]
+#[serde(untagged)]
```

Comment:
This will probably have to change if we want to distinguish between the future CID-based cases.